packages/gatsby/src/query/query-watcher.js

```diff
@@ -238,13 +232,18 @@ const watch = async rootDir => {
   })
 
   watcher = chokidar
-    .watch([
-      slash(path.join(rootDir, `/src/**/*.{js,jsx,ts,tsx}`)),
-      ...packagePaths,
-    ])
+    .watch(
+      [slash(path.join(rootDir, `/src/**/*.{js,jsx,ts,tsx}`)), ...packagePaths],
+      { ignoreInitial: true }
+    )
     .on(`change`, path => {
-      report.pendingActivity({ id: `query-extraction` })
-      debounceCompile()
+      emitter.emit(`QUERY_FILE_CHANGED`, path)
+    })
+    .on(`add`, path => {
+      emitter.emit(`QUERY_FILE_CHANGED`, path)
+    })
+    .on(`unlink`, path => {
+      emitter.emit(`QUERY_FILE_CHANGED`, path)
     })
```

Comment:
I didn't want to depend on the state machine anywhere else, partly because we're not using it for `build` right now, and when we do it'll be a different machine. Same applies to inc builds. The ideal (and something I'd do in future) is to move the watcher inside the machine. I've been looking at how that would work with webpack events too, as they're duplicated in most cases.